core/base/src/TSystem.cxx

```diff
@@ -4340,14 +4347,30 @@ TString TSystem::SplitAclicMode(const char *filename, TString &aclicMode,
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-/// Remove the shared libs produced by the CompileMacro() function.
+/// Remove the shared libs produced by the CompileMacro() function, together
+/// with their rootmaps, linkdefs, and pcms (and some more on Windows).
 
 void TSystem::CleanCompiledMacros()
 {
    TIter next(fCompiled);
    TNamed *lib;
+   const char *extensions[] = {".lib", ".exp", ".d", ".def", ".rootmap", "_ACLiC_linkdef.h", "_ACLiC_dict_rdict.pcm"};
    while ((lib = (TNamed*)next())) {
-      if (lib->TestBit(kMustCleanup)) Unlink(lib->GetTitle());
+      if (lib->TestBit(kMustCleanup)) {
+         TString libname = lib->GetTitle();
+#ifdef WIN32
+         // On Windows, we need to unload the dll before deleting it
+         if (gInterpreter->IsLibraryLoaded(libname))
+            ::FreeLibrary(::GetModuleHandle(libname));
```

Comment:
`TSystem::Unload` does more than just unloading the DLL, and for some reason it crashes in the interpreter when calling it from the `TROOT` destructor. And here we just need to unload the DLL

Reply:
> `TSystem::Unload` does more than just unloading the DLL

If `CleanCompiledMacros` is called outside of the destruction, then we would need those.

> and for some reason it crashes in the interpreter when calling it from the TROOT destructor.

Since the libraries are in the process of being unloaded, do we need to 'early' unload it here?

Reply:
> Since the libraries are in the process of being unloaded, do we need to 'early' unload it here?

Yes, otherwise it's impossible to delete the file (DLL)

Reply:
Fair enough.   Thanks.